tool/microkit/src/main.rs

```diff
@@ -4,6 +4,9 @@
 // SPDX-License-Identifier: BSD-2-Clause
 //
 
+// we want our asserts, even if the compiler figures out they hold true already during compile-time
+#![allow(clippy::assertions_on_constants)]
```

Comment:
I would be happy to move the asserts on constants outside of runtime functions but I could not find a way to do compile time asserts in Rust. Do you know how? Everything I see online tells me to download a crate....

Reply:
According to https://github.com/rust-lang/rust-clippy/issues/8159 there is `const _: () = assert!(usize::BITS >= 32);` to denote that the assert is const. Grasping from the comments it did however not work for complext expression, i. e. 

```rust
const _: () = {
    assert!(X1P_20 <= 1.0e-6); // warns :disappointed:
};
```
We can introduce that, but I'm not too fond of the additional syntax. Adding a const item that is a zero sized type only to assign the non-existant, zero-sized return value of an assert seems weird to me.